Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/polygon_soup_to_polygon_mesh.h

```diff
@@ -37,13 +38,26 @@ namespace CGAL {
 namespace Polygon_mesh_processing {
 namespace internal {
 
+template <typename PM_Point, typename PS_Point>
+PM_Point convert_to_pm_point(const PS_Point& p)
+{
+  CGAL_static_assertion((std::is_convertible<PS_Point, PM_Point>::value));
+  return PM_Point(p);
+}
+
+// just for backward compatibility reasons
+template <typename PM_Point, typename PS_FT>
+PM_Point convert_to_pm_point(const std::array<PS_FT, 3>& p)
+{
+  return PM_Point(p[0], p[1], p[2]);
```

Comment:
why not always using this one?

Reply:
* This is weaker than the generic one using explicit conversion;
* `operator[]` is in `CGAL::Point_3` but is not part of its concept;
* Might want to do weird stuff with `CGAL::Point_2` for example.